src/lib/components/no-result/terra-no-result-notice.component.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { TerraNoResultNoticeComponent } from './terra-no-result-notice.component';
+import { TerraButtonComponent } from '../..';
+import { By } from '@angular/platform-browser';
+import { noResultsNoticeMockButtons } from '../../testing/mock-buttons';
+import { Component, DebugElement } from '@angular/core';
+
+@Component({
+    selector: 'terra-button',
+    template: ''
+})
+class TerraButtonMockComponent extends TerraButtonComponent {}
+
+describe('TerraNoResultNoticeComponent', () => {
+    let fixture: ComponentFixture<TerraNoResultNoticeComponent>;
+    let component: TerraNoResultNoticeComponent;
+
+    beforeEach(() => {
+        fixture = TestBed.configureTestingModule({
+            declarations: [TerraButtonMockComponent, TerraNoResultNoticeComponent]
+        }).createComponent(TerraNoResultNoticeComponent);
+        component = fixture.componentInstance;
+
+        fixture.detectChanges();
+    });
+
+    it('should create an instance', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should highlight inner buttons', () => {
+        component.inputButtons = noResultsNoticeMockButtons;
+        fixture.detectChanges();
+        const buttons: Array<DebugElement> = fixture.debugElement.queryAll(By.directive(TerraButtonComponent));
+        expect(
+            buttons.every(
+                (button: DebugElement) => (button.componentInstance as TerraButtonComponent).inputIsHighlighted
+            )
+        ).toBe(true);
+    });
+
+    it('should ensure that inner buttons are not small', () => {
+        component.inputButtons = noResultsNoticeMockButtons;
+        fixture.detectChanges();
+        const buttons: Array<DebugElement> = fixture.debugElement.queryAll(By.css('terra-button'));
+        let buttonComponent: TerraButtonComponent = buttons[0].componentInstance as TerraButtonComponent;
+        fixture.detectChanges();
+        expect(buttonComponent.inputIsSmall).toBe(false);
```

Comment:
Du kannst hier das gleiche wie im Testfall zuvor anwenden:
```suggestion
        const buttons: Array<DebugElement> = fixture.debugElement.queryAll(By.directive(TerraButtonComponent));
        expect(
            buttons.every(
                (button: DebugElement) => !(button.componentInstance as TerraButtonComponent).inputIsSmall
            )
        ).toBe(true);
```

Reply:
ich verstehe die Aussage nicht. es müsste entweder `.not.toBe(true)` oder `.toBe(false)` sein. es macht für mich keinen Sinn, den Wert zu negieren und dann zu testen. aber es kommt halt true heraus, wenn ich es ohne das ! mache

Reply:
Du möchtest überprüfen ob `inputIsSmall` bei allen (**every**) buttons `false` (oder **nicht** gesetzt) ist. Das Ergebnis der Überprüfung sollte also "Ja" (`true`) sein.

Reply:
Nur als Randnotiz.
Die Logik die du im Kopf hast kann man ebenfalls abbilden, mit [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some):
```
const buttons: Array<DebugElement> = fixture.debugElement.queryAll(By.directive(TerraButtonComponent));
expect(buttons.some(
     (button: DebugElement) => (button.componentInstance as TerraButtonComponent).inputIsSmall
)).toBe(false);
```
Als Frage: Gibt es einen button der small ist? Erwartung: Ich hoffe nicht 😋